src/index.ts

```diff
@@ -195,6 +319,86 @@ class ComputerSpeakers implements Accessory {
       })
   }
 
+  private adjustVolume(
+    logarithmic: boolean,
+    delta: number,
+    delay: number,
+    on: boolean,
+    callback: () => void
+  ) {
+    if (on) {
+      this.log.debug(`Adjusting volume`)
+
+      this.getSystemVolume()
+        .then((homekitVolume) => {
+          let volume = logarithmic
+            ? Math.pow(10, homekitVolume / (100 / Math.log10(101))) - 1
+            : homekitVolume
+          this.log.debug(`Got current volume: ${homekitVolume}%`)
+          if (logarithmic) {
+            this.log.debug(`Converted current volume to: ${volume}%`)
+          }
+          volume = volume + delta
+          volume = Math.max(0, Math.min(100, volume))
+          this.log.debug(`Calculated adjusted volume: ${volume}%`)
+          if (!!this.fanService) {
+            this.fanService
+              .getCharacteristic(Characteristic.RotationSpeed)
+              .updateValue(Math.round(volume))
+          }
+          if (!!this.lightService) {
+            this.lightService
+              .getCharacteristic(Characteristic.Brightness)
+              .updateValue(Math.round(volume))
+          }
+          volume = logarithmic
+            ? Math.log10(1 + volume) * (100 / Math.log10(101))
+            : volume
+          if (!this.cached && volume === homekitVolume) {
+            if (volume < 100 && delta > 0) {
+              volume++
+            } else if (volume > 0 && delta < 0) {
+              volume--
+            }
+          }
+          volume = Math.max(0, Math.min(100, volume))
+          if (logarithmic) {
+            this.log.debug(`Converted adjusted volume to: ${volume}%`)
+          }
+          this.log.debug(`Being requested to set volume to ${volume}%`)
+          return this.setSystemVolume(volume)
+        })
+        .then(() => {
+          this.log.debug(`Successfully adjusted volume`)
+          callback()
+          setTimeout(
+            function () {
+              this.increaseVolumeSwitch
+                .getCharacteristic(Characteristic.On)
+                .updateValue(false)
+              this.decreaseVolumeSwitch
+                .getCharacteristic(Characteristic.On)
+                .updateValue(false)
```

Comment:
Since this is setup with `.on(CharacteristicEventTypes.GET, this.showAlwaysOff.bind(this))` does updating the value ever do anything? Does the switch show as on for `delay`ms after being toggled?

Reply:
If you switch on one of the delta switches, then the volume is increased or decreased and after `delay` ms the switch is turned off automatically.  Furthermore, the setup with `.on(CharacteristicEventTypes.GET, this.showAlwaysOff.bind(this))` is needed when the value of one of the switches is to be read from HomeKit.  Since the switches should be stateless, the value `false` is always returned, i.e., the switches are always off except in the `delay` ms after it has been turned on.